### PR TITLE
Openapi endpoint config tests

### PR DESCRIPTION
**Pull request by Azquelt**
For #25437

### CODE REVIEW

**Comment by Azquelt** [comment]
Tests show a couple of issues:
1. The OpenAPI document servlet has a hard-coded check that `/openapi` is being requested
2. When a WAB context root is updated, the bundle containing it is restarted. In the case of the OpenAPI document servlet, it's in the same bundle as the `ApplicationRegistry`. Having that bundle restarted causes the `ApplicationRegistry` to forget which applications are running and the OpenAPI documents generated for them.

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#CMVCPATHNAME io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties
+#ISMESSAGEFILE FALSE
+#NLS_ENCODING=UNICODE
+#NLS_MESSAGEFORMAT_NONE
+
+mpOpenAPI=MicroProfile OpenAPI
+mpOpenAPI.desc=Configuration for MicroProfile OpenAPI.
+
+docPath=MicroProfile OpenAPI document endpoint path
+docPath.desc=Path to obtain OpenAPI 3.0/Swagger documents
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties:23]: do this line and line 26need a period, similar to line 20? Technically none of them are complete sentences, so you could remove the period at line 20.

Also wondering if there is any additional information about specifying the path that might help the developer- as is the descriptions just restate the information in the definition

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 2]
```diff
@@ -98,3 +98,35 @@ OPENAPI_MERGE_INVALID_NAME_CWWKO1666W.useraction=Review the error text to identi
 OPENAPI_MERGE_UNUSED_INCLUDE_CWWKO1667W=CWWKO1667W: The {0} configuration property includes "{1}" but that does not match any deployed application or web module.
 OPENAPI_MERGE_UNUSED_INCLUDE_CWWKO1667W.explanation=The server is configured to create OpenAPI documentation for the application or module name, but no application or web module with that name was deployed.
 OPENAPI_MERGE_UNUSED_INCLUDE_CWWKO1667W.useraction=Check that all applications started correctly and that the names in the configuration property are specified correctly.
+
+OPEN_API_UI_PATH_UPDATE_CWWKO1668I=CWWKO1668I: The OpenAPI UI path has been set to {0}.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops:102]:
```suggestion
OPEN_API_UI_PATH_UPDATE_CWWKO1668I=CWWKO1668I: The OpenAPI UI path is set to {0}.
```

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 3]
```diff
@@ -98,3 +98,35 @@ OPENAPI_MERGE_INVALID_NAME_CWWKO1666W.useraction=Review the error text to identi
...
+OPEN_API_UI_PATH_UPDATE_CWWKO1668I.explanation=The OpenAPI UI path has been set to a value that is not the default.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops:103]:
```suggestion
OPEN_API_UI_PATH_UPDATE_CWWKO1668I.explanation=The OpenAPI UI path is set to a value that is not the default.
```

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 4]
```diff
@@ -98,3 +98,35 @@ OPENAPI_MERGE_INVALID_NAME_CWWKO1666W.useraction=Review the error text to identi
...
+OPEN_API_UI_PATH_UPDATE_CWWKO1668I.useraction=No action is required.
+
+OPEN_API_DOC_PATH_UPDATE_CWWKO1669I=CWWKO1669I: The OpenAPI Document path has been set to {0}.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops:106]:
```suggestion
OPEN_API_DOC_PATH_UPDATE_CWWKO1669I=CWWKO1669I: The OpenAPI Document path is set to {0}.
```

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 5]
```diff
@@ -98,3 +98,35 @@ OPENAPI_MERGE_INVALID_NAME_CWWKO1666W.useraction=Review the error text to identi
...
+OPEN_API_DOC_PATH_UPDATE_CWWKO1669I.explanation=The OpenAPI Document path has been set to a value that is not the default.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops:107]:
```suggestion
OPEN_API_DOC_PATH_UPDATE_CWWKO1669I.explanation=The OpenAPI Document path is set to a value that is not the default.
```

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 6]
```diff
@@ -98,3 +98,35 @@ OPENAPI_MERGE_INVALID_NAME_CWWKO1666W.useraction=Review the error text to identi
...
+OPEN_API_DOC_PATH_UPDATE_CWWKO1669I.useraction=No action is required.
+
+OPEN_API_UI_PATH_UPDATE_FAILED_CWWKO1670E=CWWKO1670E: The OpenAPI UI path is invalid. The default value will be used instead. Check previous warnings for the cause of validation failure.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops:110]:
```suggestion
OPEN_API_UI_PATH_UPDATE_FAILED_CWWKO1670E=CWWKO1670E: The OpenAPI UI path is invalid. The default value is used instead. Check previous warnings for the cause of validation failure.
```

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 7]
```diff
@@ -98,3 +98,35 @@ OPENAPI_MERGE_INVALID_NAME_CWWKO1666W.useraction=Review the error text to identi
...
+OPEN_API_UI_PATH_UPDATE_FAILED_CWWKO1670E.explanation=The OpenAPI UI path is invalid. The default value will be used instead.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops:111]:
```suggestion
OPEN_API_UI_PATH_UPDATE_FAILED_CWWKO1670E.explanation=The OpenAPI UI path is invalid. The default value is used instead.
```

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 8]
```diff
@@ -98,3 +98,35 @@ OPENAPI_MERGE_INVALID_NAME_CWWKO1666W.useraction=Review the error text to identi
...
+OPEN_API_UI_PATH_UPDATE_FAILED_CWWKO1670E.useraction=Check earlier error messages for the cause of validation failure.
+
+OPEN_API_DOC_PATH_UPDATE_FAILED_CWWKO1671E=CWWKO1671E: The OpenAPI Document path is invalid. The default value will be used instead. Check previous warnings for the cause of validation failure.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops:114]:
```suggestion
OPEN_API_DOC_PATH_UPDATE_FAILED_CWWKO1671E=CWWKO1671E: The OpenAPI Document path is invalid. The default value is used instead. Check previous warnings for the cause of validation failure.
```

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 9]
```diff
@@ -98,3 +98,35 @@ OPENAPI_MERGE_INVALID_NAME_CWWKO1666W.useraction=Review the error text to identi
...
+OPEN_API_DOC_PATH_UPDATE_FAILED_CWWKO1671E.explanation=The OpenAPI Document path is invalid. The default value will be used instead.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops:115]:
```suggestion
OPEN_API_DOC_PATH_UPDATE_FAILED_CWWKO1671E.explanation=The OpenAPI Document path is invalid. The default value is used instead.
```

**Review by dmuelle — COMMENTED** [review]
ID review- minor edits to avoid past perfect and future tenses in messages per IBM style

**Comment by Azquelt** [comment]
Sorry, @dmuelle, this PR does not need messages reviewed.

The messages changes are from #25436 and will disappear from this PR once that one is merged. (We're having some problems getting it through a build at the moment).

I'd opened it as a draft and hadn't realised libby would tag you for review. I should have added the review not needed tags.

**Comment by Azquelt** [comment]
I will apply @dmuelle's suggestions to #25436.

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (Azquelt) [reply dev/io.openliberty.microprofile.openapi.internal.common/resources/OSGI-INF/l10n/metatype.properties:23]: Looking at the existing docs, it looks like our exiting descriptions do end with a period, even when it's not a full sentence so I'll add one here.

I agree the descriptions should be more helpful, opened #25559

Changes will go into #25436

File: dev/com.ibm.ws.microprofile.openapi_fat/fat/src/com/ibm/ws/microprofile/openapi/fat/config/OpenAPIConfigTest.java [hunk 10]
```diff
@@ -0,0 +1,421 @@
...
+    public void testApplicationPathConfict() throws Exception {
+        // Test if initial config has a conflict - as Test app starts last, expect it to fail to start
+        ServerConfiguration config = server.getServerConfiguration();
+        config.getMpOpenAPIElement().setDocPath(null);
+        config.getMpOpenAPIElement().setUiPath("/mpOpenAPIConfigTest");
+        server.updateServerConfiguration(config);
+
+        server.startServer(false);
+
+        assertNotNull("Web Application fails to start due to context path conflict with OPENAPIUI bundle", "SRVE0164E.*OpenAPIUI");
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.openapi_fat/fat/src/com/ibm/ws/microprofile/openapi/fat/config/OpenAPIConfigTest.java]: Not actually checking the logs here.

File: dev/com.ibm.ws.microprofile.openapi_fat/fat/src/com/ibm/ws/microprofile/openapi/fat/config/OpenAPIConfigTest.java [hunk 11]
```diff
@@ -0,0 +1,421 @@
...
+        server.startServer(false);
+
+        assertNotNull("Document Web Appplication path contains invalid characters",
+                server.waitForStringInLog("CWWKO1676E.*$")); // check that error indicating that conflict is thrown
+        assertNotNull("Document Web Appplication path is invalid",
+                server.waitForStringInLog("CWWKO1671E.*$")); // check that error indicating that conflict is thrown
+        assertNotNull("UI Web Appplication path contains invalid characters",
+                server.waitForStringInLog("CWWKO1675E.*$")); // check that error indicates invalid characters is logged
+        assertNotNull("UI Web Appplication path is invalid",
+                server.waitForStringInLog("CWWKO1670E.*$")); // check that error indicating that a failure has occurred has been thrown
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.openapi_fat/fat/src/com/ibm/ws/microprofile/openapi/fat/config/OpenAPIConfigTest.java]: the `.*$` isn't necessary here

**Comment by Azquelt** [comment]
These changes have been rolled into https://github.com/OpenLiberty/open-liberty/pull/25436

**Pull request closed without merging**